bin/test-external-partner-project.js

```diff
@@ -109,34 +120,43 @@ try {
 try {
   debug('Preparing Package To Run Tests Against Commit');
   insertTarballsToPackageJson(packageJsonLocation);
-  // we must use npm because yarn fails to install
-  // the nested tarballs correctly (just as it fails to pack them correctly)
-  // we rimraf node_modules first because it was previously
-  // installed using yarn's resolution algorithm
-  if (!skipSmokeTest) {
-    execExternal(`rm -rf node_modules`);
-  }
-  execExternal(`npm install`);
+
+  // clear node_modules installed for the smoke-test
+  execExternal(`rm -rf node_modules`);
+  // we are forced to use yarn so that our resolutions will be respected
+  // in addition to the version file link we insert otherwise nested deps
+  // may bring their own ember-data
+  //
+  // For this reason we don't trust the lock file
+  // we also can't trust the cache
+  execExternal(`yarn install --no-lockfile --cache-folder=tmp/yarn-cache`);
```

Comment:
FWIW, npm and yarn aren't drop in replacements for each other. There are quite a few bizarre inconsistencies (e.g. around using `file:` or `link:` references for deps/devDeps).

Reply:
Before we were using npm regardless of whether the partner used yarn or npm because yarn was failing to extract the tarballs. This appears to have been fixed! However, I discovered a bug with npm wherein both during link and when installing from tarball nested dependencies specifying ember-data wont resolve to our tarball even if it is "in range" (I think this may be due to it being an alpha).

So we can't use npm safely, so using yarn+resolutions (to force the nested deps) always seems like the way to go.

Reply:
FWIW, tarballs can never be "in range". They are always treated as "exotic" and the system cannot make assumptions about greater than / less than.

Reply:
@rwjblue all the more reason to not install with npm unless they’ve added support for resolutions that I don’t know about (based on testing they don’t seem to respect them)